### DISCUSSION TITLE
Modelling a deformable softbody gripper in pyBullet

### DISCUSSION THREAD

**Discussion by achuwilson** [bulletphysics/bullet3#3569] [General]
Hi, I was wondering whether a deformable gripper like the GelSight gripper (linked below)can be simulated in pyBullet. 

[MIT News on GelSight](http://news.mit.edu/2017/gelsight-robots-sense-touch-0605)
[Arxiv Paper on GelSight](https://arxiv.org/abs/1708.00922)

I tried to model a tactile sensor using the getContactPoints API, but it returns only a couple of contact points when gripping objects, which may be attributed to the non-deformability of the solids. Is it possible to expose the softbody simulation capabilities of Bullet in python APIs. In addition, will the collision detection queries work with the softbodies.

**Comment by ashwal** [comment]
You can take a look at issue #1432 and #1175 for some conversation around this, but no outward facing progress has been made.

I'm working on basic bindings right now.

**Comment by JanMatas** [comment]
HI @ashwal 

have you made any progress on the bindings lately? I am looking for a way to simulate cloth grasping using pybullet. 

Thanks!

**Comment by ashwal** [comment]
@JanMatas For my uses it ended up being faster just to write in C++ in hacky way. Sorry!

**Comment by ghost** [comment]
Hi @ashwal 
could you just lay out in 5 steps how to do so?
New to python here :)
Thank you!

**Comment by aprath1** [comment]
Hello,
Any developments in this direction? I am working on simulating a gripper which is non rigid and could imitate a silicone rubber like property, (even if not exactly same but some deformable property would be good).
@erwincoumans - I think this feature would be a nice addition to robotic research field and many others... Could you please let me know if it is possible to make a robot gripper as a soft body? This would mean that the left and right gripper links (for a 2 finger gripper) has to be a soft body... and then use it to grasp rigid bodies..

I saw that simulating a soft body examples but is it possible to load from just .obj files or .vtk format is also needed? For example the ball object (ballId_1) loads properly with .vtk file. On the other hand the ball object (ballId_2) shows unstable weird behavior when only .obj is used, please see the image and code below... 
`ballId_1 = p.loadSoftBody("ball.obj",  basePosition = [0,0, 2.5], scale = 0.5, mass = 4, useNeoHookean = 1, NeoHookeanMu = 400, NeoHookeanLambda = 600, NeoHookeanDamping = 0.001, useSelfCollision = 1, frictionCoeff = .5, collisionMargin = 0.001)`

`ballId_2 = p.loadSoftBody("ball.obj",  simFileName = "ball.vtk", basePosition = [0,0, 2.5], scale = 0.5, mass = 4, useNeoHookean = 1, NeoHookeanMu = 400, NeoHookeanLambda = 600, NeoHookeanDamping = 0.001, useSelfCollision = 1, frictionCoeff = .5, collisionMargin = 0.001)`

![Soft_body_issue_1](https://user-images.githubusercontent.com/88531660/144417379-674cf74c-5a85-4525-b6b4-40bd68d84a40.jpg)

**Comment by DanielTakeshi** [comment]
You might be interested in TACTO. https://github.com/facebookresearch/tacto
